src/goto-programs/goto_convert_functions.cpp

```diff
@@ -166,10 +166,21 @@ void goto_convert_functionst::convert_function(
   // we have a body, make sure all parameter names are valid
   for(const auto &p : f.parameter_identifiers)
   {
-    DATA_INVARIANT(!p.empty(), "parameter identifier should not be empty");
-    DATA_INVARIANT(
+    DATA_INVARIANT_WITH_DIAGNOSTICS(
+      !p.empty(),
+      "parameter identifier should not be empty",
+      "function:",
+      identifier,
+      "parameter:",
+      p);
+
```

Comment:
That doesn't make sense: when the message is triggered, `p` is empty. Thus, why log it?

Reply:
Right. Removed `p` from this case.